### PR TITLE
Updates for vllm 0.6.2

### PR DESCRIPTION
**Pull request by gc-fu**
## Description

Updates for vLLm to using vLLM 0.6.2.

We need to change the followings:

- [x] Initial Dockerfile
- [x] vLLM related updates
- [x] update benchmark_latency.py 
- [x] update benchmark_throughput.py
- [x] Examples, in `ipex-llm/python/llm/example/GPU/vLLM-Serving`
- [x] vLLM worker
- [x] Update final Dockerfile before merge, this is for changing building branches, check the TODO in the code.
- [x] Merge https://github.com/analytics-zoo/vllm/pull/47
- [x] Test image functionality... Done by Wang, Jun

### 4. How to test?
- [ ] N/A
- [x] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. Known issues

- [x] Sometimes, this will fail on initial start up, and got timeout error...

### CODE REVIEW

**Comment by gc-fu** [comment]
https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/11699472618

File: python/llm/example/GPU/vLLM-Serving/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ In this example, we will run Llama2-7b model using Arc A770 and provide `OpenAI-
 
 ### 0. Environment
 
-To use Intel GPUs for deep-learning tasks, you should install the XPU driver and the oneAPI Base Toolkit 2024.0. Please check the requirements at [here](https://github.com/intel-analytics/ipex-llm/tree/main/python/llm/example/GPU#requirements).
+To use Intel GPUs for deep-learning tasks, you should install the XPU driver and the oneAPI Base Toolkit 2024.1. Please check the requirements at [here](https://github.com/intel-analytics/ipex-llm/tree/main/python/llm/example/GPU#requirements).
```
Comment (xiangyuT) [inline python/llm/example/GPU/vLLM-Serving/README.md:20]: Is the link correspond with oneapi 2024.1? ipex-llm is mainly using 2024.0 on arc?

File: python/llm/example/GPU/vLLM-Serving/README.md [hunk 2]
```diff
@@ -44,14 +44,13 @@ conda create -n ipex-vllm python=3.11
...
-git clone -b sycl_xpu https://github.com/analytics-zoo/vllm.git
+# TODO: check this later... for the specific branch
+git clone -b  https://github.com/analytics-zoo/vllm.git
 cd vllm
-pip install -r requirements-xpu.txt
-pip install --no-deps xformers
-VLLM_BUILD_XPU_OPS=1 pip install --no-build-isolation -v -e .
-pip install outlines==0.0.34 --no-deps
-pip install interegular cloudpickle diskcache joblib lark nest-asyncio numba scipy
+VLLM_TARGET_DEVICE=xpu pip install --no-build-isolation -v . && \
```
Comment (xiangyuT) [inline python/llm/example/GPU/vLLM-Serving/README.md:53]: remove '&& \'

File: python/llm/example/GPU/vLLM-Serving/README.md [hunk 3]
```diff
@@ -61,6 +60,7 @@ pip install transformers_stream_generator einops tiktoken
 ```bash
 export USE_XETLA=OFF
 export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
```
Comment (xiangyuT) [inline python/llm/example/GPU/vLLM-Serving/README.md:62]: export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=2 in `start-vllm-service.sh`. Should we align this?

File: python/llm/example/GPU/vLLM-Serving/README.md (hunk 1 above)
Comment (gc-fu) [reply python/llm/example/GPU/vLLM-Serving/README.md:20]: Fixed using link:https://www.intel.com/content/www/us/en/docs/oneapi/installation-guide-linux/2024-1/overview.html

**Comment by gc-fu** [comment]
https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/11793680121

**Review by xiangyuT — APPROVED** [review]
lgtm

**Pull request merged**